cmd/dep/init.go

```diff
@@ -180,6 +180,17 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	}
 	l = dep.LockFromInterface(soln)
 
+	// Find any solved dependencies that are unconstrained and remove them from
+	// the manifest.
+	for pr, d := range m.Dependencies {
+		if d.Constraint == nil {
+			if ctx.Loggers.Verbose {
+				ctx.Loggers.Err.Printf("dep: Removing unconstrained dependency \"%v\" from manifest.\n", pr)
+			}
+			delete(m.Dependencies, pr)
+		}
+	}
+
```

Comment:
Nice, this would do. But I see projects are added to manifest from `projectData`'s `constraints` [here](https://github.com/golang/dep/blob/master/cmd/dep/init.go#L124), which gets filled up [here](https://github.com/golang/dep/blob/master/cmd/dep/init.go#L410). I think if you perform a similar check on the value returned by `getProjectPropertiesFromVersion()`, we can avoid adding the project to manifest at all.